backend/src/services/notifications.ts

```diff
@@ -0,0 +1,132 @@
+  schedule.scheduleJob("*/1 * * * *", async () => {
+    try {
+      const upcomingNotifSessions = await Session.find({
+        upcomingSessionNotifSent: { $eq: false },
+      });
+      upcomingNotifSessions.forEach(async (session) => {
+        const dateNow = new Date();
+        const mentee = await Mentee.findById(session.menteeId);
+        if (!mentee) {
+          throw InternalError.ERROR_GETTING_MENTEE;
+        }
+        const mentor = await Mentor.findById(session.mentorId);
+        if (!mentor) {
+          throw InternalError.ERROR_GETTING_MENTOR;
+        }
+        if (session.startTime.getTime() - dateNow.getTime() <= 86400000) {
+          if (session.preSessionCompleted) {
+            const menteeNotif = await sendNotification(
+              "You have an upcoming session.",
+              `Ready for your session with ${mentor.name} in 24 hours? \u{1F440}`,
+              mentee.fcmToken
+            );
+            console.log("Function executed successfully:", menteeNotif);
+            const mentorNotif = await sendNotification(
+              "You have an upcoming session.",
+              `Ready for your session with " + mentee.name + " in 24 hours? \u{1F440}. Check out " + mentee.name + "'s pre-session notes.`,
+              mentor.fcmToken
+            );
+            console.log("Function executed successfully:", mentorNotif);
+          } else {
+            const menteeNotif = await sendNotification(
+              "You have an upcoming session.",
+              `Ready for your session with " + mentor.name + " in 24 hours? \u{1F440}. Fill out your pre-session notes now!`,
+              mentee.fcmToken
+            );
+            console.log("Function executed successfully:", menteeNotif);
+            const mentorNotif = await sendNotification(
+              "You have an upcoming session.",
+              `Ready for your session with ${mentee.name} in 24 hours? \u{1F440}`,
+              mentor.fcmToken
+            );
+            console.log("Function executed successfully:", mentorNotif);
+          }
+          session.upcomingSessionNotifSent = true;
+          await session.save();
+        }
+      });
+    } catch (error) {
+      console.error("Error executing function:", error);
+    }
+  });
+}
+
+async function startPostSessionCronJob() {
+  schedule.scheduleJob("*/1 * * * *", async () => {
+    try {
+      const postNotifSessions = await Session.find({ postSessionNotifSent: { $eq: false } });
+      postNotifSessions.forEach(async (session) => {
+        const dateNow = new Date();
+        const mentee = await Mentee.findById(session.menteeId);
+        if (!mentee) {
+          throw InternalError.ERROR_GETTING_MENTEE;
+        }
+        const mentor = await Mentor.findById(session.mentorId);
+        if (!mentor) {
+          throw InternalError.ERROR_GETTING_MENTOR;
+        }
+        if (dateNow.getTime() - session.endTime.getTime() >= 0) {
```

Comment:
Lets not send it immediately after the session ends. Maybe give some buffer time like 15-30 mins.